Week3/homework/server.js

```diff
@@ -0,0 +1,190 @@
+const express = require('express');
+const app = express();
+const parser = require('body-parser');
+const path = require('path');
+const body_parser = require('body-parser');
+const urlParser = body_parser.urlencoded({
+  extended: false,
+});
+
+const mysql = require('mysql');
+var con = mysql.createConnection({
+  host: 'localhost',
+  user: 'hyfuser',
+  password: 'hyfpassword',
+  database: 'todo',
+});
+
+app.use(express.urlencoded());
+
+con.connect();
+const port = process.env.PORT || 3000;
+
+app.set('view engine', 'ejs');
+
+app.get('/', (req, res) => {
+  res.render('index');
+});
+
+app.get('/login', (req, res) => {
+  res.render('login', { error: null });
+});
+
+app.get('/Signup', (req, res) => {
+  res.render('Signup');
+});
+app.get('/About', (req, res) => {
+  res.render('About');
+});
+app.get('/Home', (req, res) => {
+  res.render('index');
+});
+
+// sign up => add user
+app.post('/Signup', urlParser, (req, res) => {
+  con.query('select * from users', (err, result) => {
+    if (err) console.log(err);
+    con.query(
+      `insert into users values (?,?,?)`,
+      [result.length + 1, req.body.username, req.body.password],
+      (err, data) => {
+        if (err) {
+          res.render('error', { data: 'error' });
+        } else {
+          res.render('index');
+        }
+        con.end();
```

Comment:
If you close the connection then any queries run after this code won't work and will generate connection closed issue.
This comment is for all queries